main/main.go

```diff
@@ -251,21 +261,28 @@ func runRelayer(logger logging.Logger,
 				)
 				continue
 			}
-		case err := <-subscriber.Err():
+		case err := <-relayer.Subscriber.Err():
 			logger.Error(
 				"Received error from subscribed node",
 				zap.String("originChainID", sourceSubnetInfo.BlockchainID),
 				zap.Error(err),
 			)
-			err = subscriber.Subscribe()
+			err = relayer.ReconnectToSubscriber()
 			if err != nil {
 				logger.Error(
-					"Failed to resubscribe to node. Relayer goroutine exiting.",
+					"Relayer goroutine exiting.",
 					zap.String("originChainID", sourceSubnetInfo.BlockchainID),
 					zap.Error(err),
 				)
-				return
+				return fmt.Errorf("relayer goroutine exiting: %w", err)
 			}
+		case <-ctx.Done():
```

Comment:
Just to confirm my understanding: When a relayer goroutine returns an error in this select loop, the errgroup will cancel the context, and all other relayer goroutines will then be marked unhealthy here. Immediately after, the errgroup will finish waiting and the application will exit anyway.

Reply:
When a relayer goroutine exits because of an error _anywhere_, the errgroup cancels the context, and all other relayer goroutines will then be marked unhealthy and exit if they're already waiting in this select statement, or once they reach it again.